src/avif.c

```diff
@@ -179,44 +180,47 @@ avifResult avifImageCopy(avifImage * dstImage, const avifImage * srcImage, avifP
     avifRWDataSet(&dstImage->exif, srcImage->exif.data, srcImage->exif.size);
     avifImageSetMetadataXMP(dstImage, srcImage->xmp.data, srcImage->xmp.size);
 
+    assert(AVIF_CHAN_A > AVIF_CHAN_V);
+    int firstPlane = AVIF_CHAN_A; // No plane by default.
+    int lastPlane = AVIF_CHAN_V;
     if ((planes & AVIF_PLANES_YUV) && srcImage->yuvPlanes[AVIF_CHAN_Y]) {
         const avifResult allocationResult = avifImageAllocatePlanes(dstImage, AVIF_PLANES_YUV);
         if (allocationResult != AVIF_RESULT_OK) {
             return allocationResult;
         }
-
-        avifPixelFormatInfo formatInfo;
-        avifGetPixelFormatInfo(srcImage->yuvFormat, &formatInfo);
-        uint32_t uvHeight = (dstImage->height + formatInfo.chromaShiftY) >> formatInfo.chromaShiftY;
-        for (int yuvPlane = 0; yuvPlane < 3; ++yuvPlane) {
-            uint32_t planeHeight = (yuvPlane == AVIF_CHAN_Y) ? dstImage->height : uvHeight;
-
+        for (int yuvPlane = 1; yuvPlane < 3; ++yuvPlane) {
```

Comment:
> Note: I took a closer look at the code, and I believe we can delete this for loop (lines 191-199) because of the `avifImageFreePlanes(dstImage, AVIF_PLANES_ALL)` call at line 175. But that would be a cleanup change and you don't need to make that change in this pull request.

I believe this code was originally meant for badly initialized images such as only `yuvPlanes[AVIF_CHAN_V]` being `NULL` for a 4:4:4 image. `avifImageAllocatePlanes()` only cares about the `format`, so the pointers are checked separately in `avifImageCopy()`. It should be replaced with a `if (planeHeight && !srcRow) { return AVIF_RESULT_INVALID_ARGUMENT; }` in another PR.

I removed it from `avifImageCopyAndPad()`, because `avifImageCopyAndPad()` is not a public API function.